Zend/zend_language_scanner.l

```diff
@@ -109,6 +109,22 @@ do {																			\
 	} \
 }
 
+#define STRIP_UNDERSCORES(n, len) \
+{ \
+	int i, old_len = len; \
+	char *new_n, *old_n; \
+	for (i = 0, new_n = old_n = n; i < old_len; ++i, ++old_n) { \
+		if (*old_n != '_') { \
+			*new_n++ = *old_n; \
+		} else { \
+			--len; \
+		} \
+	} \
+	if (old_len > len) { \
+		*new_n = '\0'; \
+	} \
+}
```

Comment:
Please use a function instead of a macro for this.

Reply:
I replaced the macro with a function. This is my first time writing code in C, so please let me know if I should implement it differently.